src/ddmd/expression.d

```diff
@@ -15682,6 +15682,14 @@ extern (C++) final class EqualExp : BinExp
         if (e1.op == TOKtype || e2.op == TOKtype)
             return incompatibleTypes();
 
+        {
+            auto t1 = e1.type;
+            auto t2 = e2.type;
+            if (t1.ty == Tenum && t2.ty == Tenum && !t1.equivalent(t2))
+                deprecation("Comparison between different enumeration types (got %s and %s)",
```

Comment:
Please suggest using `std.conv.asOriginalType` in the error message as a means to port code forward if needed.